spacy/tests/pipeline/test_tok2vec.py

```diff
@@ -189,9 +189,8 @@ def test_tok2vec_listener(with_vectors):
             tagger.add_label(tag)
 
     # Check that the Tok2Vec component finds it listeners
-    assert tok2vec.listeners == []
-    optimizer = nlp.initialize(lambda: train_examples)
     assert tok2vec.listeners == [tagger_tok2vec]
+    optimizer = nlp.initialize(lambda: train_examples)
```

Comment:
Now the listeners are also set when the pipe is added rather than on initialize, so it's not empty before initialization anymore. I think you could remove `_link_components` from `initialize` now, but I was trying to change as little as possible for now.